File: components/maps/src/test/java/org/storm/maps/layer/BackgroundLayerTest.java
package org.storm.maps.layer;

import javafx.application.Application;
import javafx.scene.input.KeyCode;
import javafx.stage.Stage;
import lombok.extern.slf4j.Slf4j;
import org.storm.core.ui.Resolution;
import org.storm.core.ui.Resolutions;
import org.storm.core.ui.Window;

public class BackgroundLayerTest extends Application {

  @Override
  public void start(Stage primaryStage) throws Exception {
    Window window = new Window(Resolutions.HD);
    BackgroundLayer backgroundLayer = new BackgroundLayer("src/test/resources/background/testBackground.png", Resolutions.HD);

    backgroundLayer.render(window.getGraphicsContext(), 300, 200);

    primaryStage.setScene(window);
    primaryStage.show();
  }

}

File: components/maps/src/test/kotlin/org/storm/maps/layer/BackgroundLayerTest.kt
package org.storm.maps.layer

import javafx.application.Application
import javafx.stage.Stage
import kotlinx.coroutines.runBlocking
import org.storm.core.context.Context
import org.storm.core.context.setResolution
import org.storm.core.ui.Resolution
import org.storm.core.ui.impl.JfxWindow

class BackgroundLayerTest : Application() {
    override fun start(primaryStage: Stage) {
        Context.setResolution(Resolution.HD)
        val window = JfxWindow()
        val backgroundLayer = BackgroundLayer("src/test/resources/background/testBackground.png")

        runBlocking { backgroundLayer.render(window.canvas, 300.0, 200.0) }

        primaryStage.scene = window
        primaryStage.show()
    }

}
